core/auth/src/main/java/software/amazon/awssdk/auth/credentials/InstanceProfileCredentialsProvider.java

```diff
@@ -273,7 +273,14 @@ private String handleTokenErrorResponse(Exception e) {
     }
 
     private boolean isInsecureFallbackDisabled() {
-        return ec2MetadataDisableV1Resolver.resolve();
+        if (isInsecureFallbackDisabled == null) {
+            synchronized (lock) {
```

Comment:
nit: we can use Lazy<Boolean> so we don't need to write this logic https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/Lazy.java

Reply:
Good suggestion, I had forgotten about the Lazy